cosmos-server/src/main/scala/com/mesosphere/cosmos/converter/Universe.scala

```
@@ -1,86 +1,308 @@
+          postUninstallNotes = value.postUninstallNotes,
+          licenses = value.licenses.as[Option[List[universe.v2.model.License]]]
+      )
+    } { _ =>
+      Try(throw new IllegalArgumentException(
+              "Cannot convert version 2 package details to version 3 package"))
+    }
+  }
 
-    Bijection.build(fwd)(rev)
+  implicit val v3V2PackageToInternalPackageDefinition: Injection[
+      universe.v3.model.V2Package, internal.model.PackageDefinition] = {
+    Injection.build { (value: universe.v3.model.V2Package) =>
+      internal.model.PackageDefinition(
+          value.packagingVersion,
+          value.name,
+          value.version,
+          value.releaseVersion,
+          value.maintainer,
+          value.description,
+          value.tags,
+          value.selected,
+          value.scm,
+          value.website,
+          value.framework,
+          value.preInstallNotes,
+          value.postInstallNotes,
+          value.postUninstallNotes,
+          value.licenses,
+          None,
+          Some(value.marathon),
+          value.resource.as[Option[universe.v3.model.V3Resource]],
+          value.config,
+          value.command
+      )
+    } { (value: internal.model.PackageDefinition) =>
+      // TODO(version): deal with minDcosReleaseVersion
+      for {
+        packagingVersion <- Try(value.packagingVersion.asInstanceOf[
+                                   universe.v3.model.V2PackagingVersion])
+        resource <- value.resource
+                     .as[Try[Option[universe.v3.model.V2Resource]]]
+      } yield {
+        universe.v3.model.V2Package(
+            packagingVersion,
+            value.name,
+            value.version,
+            value.releaseVersion,
+            value.maintainer,
+            value.description,
+            value.marathon.getOrElse(universe.v3.model
+                  .Marathon(ByteBuffer.allocate(0))), // TODO(version): Is this correct?
```

This is similar to the situation in `UniverseClient`, no? In v2 packages, the `marathon.json.mustache` file is required, so if one doesn't exist it's an error. So we either need to throw an error here or make `V2Package` more lenient with an `Option[Marathon]`.